google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/JsonStreamWriter.java

```diff
@@ -0,0 +1,384 @@
+     *     "projects/[^/]+/datasets/[^/]+/tables/[^/]+/streams/[^/]+"
+     * @param tableSchema schema used to convert Json to proto messages.
+     * @param client
+     */
+    private Builder(String streamName, Table.TableSchema tableSchema, BigQueryWriteClient client) {
+      this.streamName = streamName;
+      this.tableSchema = tableSchema;
+      this.client = client;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's TransportChannelProvider.
+     *
+     * @param channelProvider
+     * @return Builder
+     */
+    public Builder setChannelProvider(TransportChannelProvider channelProvider) {
+      this.channelProvider =
+          Preconditions.checkNotNull(channelProvider, "ChannelProvider is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's CredentialsProvider.
+     *
+     * @param credentialsProvider
+     * @return Builder
+     */
+    public Builder setCredentialsProvider(CredentialsProvider credentialsProvider) {
+      this.credentialsProvider =
+          Preconditions.checkNotNull(credentialsProvider, "CredentialsProvider is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's BatchingSettings.
+     *
+     * @param batchingSettings
+     * @return Builder
+     */
+    public Builder setBatchingSettings(BatchingSettings batchingSettings) {
+      this.batchingSettings =
+          Preconditions.checkNotNull(batchingSettings, "BatchingSettings is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's RetrySettings.
+     *
+     * @param retrySettings
+     * @return Builder
+     */
+    public Builder setRetrySettings(RetrySettings retrySettings) {
+      this.retrySettings = Preconditions.checkNotNull(retrySettings, "RetrySettings is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's ExecutorProvider.
+     *
+     * @param executorProvider
+     * @return Builder
+     */
+    public Builder setExecutorProvider(ExecutorProvider executorProvider) {
+      this.executorProvider =
+          Preconditions.checkNotNull(executorProvider, "ExecutorProvider is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's Endpoint.
+     *
+     * @param endpoint
+     * @return Builder
+     */
+    public Builder setEndpoint(String endpoint) {
+      this.endpoint = Preconditions.checkNotNull(endpoint, "Endpoint is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the action to perform when there is a schema update.
+     *
+     * @param onSchemaUpdateRunnable An abstract class that implements runnable and provides access
+     *     to the JsonStreamWriter, the StreamWriter, and the updated schema. Users should implement
+     *     the run() function.
+     * @return Builder
+     */
+    public Builder setOnSchemaUpdateRunnable(OnSchemaUpdateRunnable onSchemaUpdateRunnable) {
```

Comment:
You should define your SchemaRunnable inside JsonWriter instead of letting user to define it. User shouldn't be worried about SchemaUpdate at all.

Reply:
Got it